controllers/ingreso.go

```diff
@@ -0,0 +1,191 @@
+// @Param	limit	query	string	false	"Limit the size of result set. Must be an integer"
+// @Param	offset	query	string	false	"Start position of result set. Must be an integer"
+// @Success 200 {object} models.Ingreso
+// @Failure 404 not found resource
+// @router / [get]
+func (c *IngresoController) GetAll() {
+	var fields []string
+	var sortby []string
+	var order []string
+	var query = make(map[string]string)
+	var limit int64 = 10
+	var offset int64
+
+	// fields: col1,col2,entity.col3
+	if v := c.GetString("fields"); v != "" {
+		fields = strings.Split(v, ",")
+	}
+	// limit: 10 (default is 10)
+	if v, err := c.GetInt64("limit"); err == nil {
+		limit = v
+	}
+	// offset: 0 (default is 0)
+	if v, err := c.GetInt64("offset"); err == nil {
+		offset = v
+	}
+	// sortby: col1,col2
+	if v := c.GetString("sortby"); v != "" {
+		sortby = strings.Split(v, ",")
+	}
+	// order: desc,asc
+	if v := c.GetString("order"); v != "" {
+		order = strings.Split(v, ",")
+	}
+	// query: k:v,k:v
+	if v := c.GetString("query"); v != "" {
+		for _, cond := range strings.Split(v, ",") {
+			kv := strings.SplitN(cond, ":", 2)
+			if len(kv) != 2 {
+				c.Data["json"] = errors.New("Error: invalid query key/value pair")
+				c.ServeJSON()
+				return
+			}
+			k, v := kv[0], kv[1]
+			query[k] = v
+		}
+	}
+
+	l, err := models.GetAllIngreso(query, fields, sortby, order, offset, limit)
+	if err != nil {
+		logs.Error(err)
+		c.Data["mesaage"] = "Error service GetAll: The request contains an incorrect parameter or no record exists"
+		c.Abort("404")
+	} else {
+		if l == nil {
+			l = []interface{}{}
+		}
+		c.Data["json"] = map[string]interface{}{"Success": true, "Status": "200", "Message": "Request successful", "Data": l}
+	}
+	c.ServeJSON()
+}
+
+// Put ...
+// @Title Put
+// @Description update the Ingreso
+// @Param	id		path 	string	true		"The id you want to update"
+// @Param	body		body 	models.Ingreso	true		"body for Ingreso content"
+// @Success 200 {object} models.Ingreso
+// @Failure 400 the request contains incorrect syntax
+// @router /:id [put]
+func (c *IngresoController) Put() {
+	idStr := c.Ctx.Input.Param(":id")
+	id, _ := strconv.Atoi(idStr)
+	v := models.Ingreso{Id: id}
+	if err := json.Unmarshal(c.Ctx.Input.RequestBody, &v); err == nil {
+		if err := models.UpdateIngresoById(&v); err == nil {
+			c.Data["json"] = map[string]interface{}{"Success": true, "Status": "200", "Message": "Update successful", "Data": v}
+		} else {
+			logs.Error(err)
+			c.Data["mesaage"] = "Error service Put: The request contains an incorrect data type or an invalid parameter"
+			c.Abort("400")
+		}
+	} else {
+		logs.Error(err)
+		c.Data["mesaage"] = "Error service Put: The request contains an incorrect data type or an invalid parameter"
+		c.Abort("400")
+	}
+	c.ServeJSON()
+}
+
+// Delete ...
+// @Title Delete
+// @Description delete the Ingreso
+// @Param	id		path 	string	true		"The id you want to delete"
+// @Success 200 {string} delete success!
```

Comment:
De manera similar a la L176 de `controllers/afectacion_presupuestal.go`
```suggestion
// @Success 200 {object} models.Eliminado
```